shell/platform/darwin/macos/framework/Source/FlutterViewController.mm

```diff
@@ -501,6 +506,14 @@ - (void)keyUp:(NSEvent*)event {
   }
 }
 
+- (void)flagsChanged:(NSEvent*)event {
+  if ((event.modifierFlags & NSEventModifierFlagDeviceIndependentFlagsMask) == 0) {
```

Comment:
So the `modifierFlags` only contains information about this current key event, and not the current modifier state, right? For instance, will this be zero if I have the control and shift keys down, and this event is a result of releasing the shift key? Or will it still have the bits for the control key set?

Reply:
Yeah I missed that. You and Stuart caught it at the same time :). Uploaded fix.